h/streamer/metrics.py

```diff
@@ -33,14 +33,15 @@ def websocket_metrics(queue):
 def metrics_process(registry, queue):  # pragma: no cover
     session = db.get_session(registry.settings)
 
-    newrelic.agent.initialize()
+    newrelic.agent.initialize(
+        config_file=resource_filename("h.streamer", "conf/newrelic.ini"),
+        log_file="debug",
+    )
+    newrelic.agent.register_application(timeout=5)
     application = newrelic.agent.application()
 
     while True:
         with db.read_only_transaction(session):
-
-            newrelic.agent.record_custom_metrics(
-                websocket_metrics(queue), application=application
-            )
+            application.record_custom_metrics(websocket_metrics(queue))
```

Comment:
This whole thing baffles me. When you look in the code there are `record_custom_metrics` on the application, agent and floating about by itself. They all seem to connect to each other in some confusing way.

I suspect we're supposed to use a `BackgroundTask` wrapper here, but that seems to add instrumentation to this code, which we don't actually want.

This seems to work though.